chain/ethereum/src/chain/mod.rs

```diff
@@ -0,0 +1,282 @@
+        .ok()
+        .map(|s| u64::from_str(&s)
+             .unwrap_or_else(|_| panic!("failed to parse env var ETHEREUM_ANCESTOR_COUNT")))
+        .unwrap_or(50);
+
+    // Default to a block polling interval of 500ms
+    static ref POLLING_INTERVAL: Duration = env::var("ETHEREUM_POLLING_INTERVAL")
+        .ok()
+        .map(|s| u64::from_str(&s).unwrap_or_else(|_| panic!("failed to parse env var ETHEREUM_POLLING_INTERVAL")))
+        .map(|n| Duration::from_millis(n))
+        .unwrap_or(Duration::from_millis(500));
+}
+
+pub struct Descriptor {
+    pub id: NetworkInstanceId,
+    pub url: String,
+}
+
+pub struct ConnectOptions<'a, MR> {
+    pub descriptor: Descriptor,
+    pub logger: &'a Logger,
+    pub metrics_registry: Arc<MR>,
+}
+
+pub struct Connection {
+    pub id: NetworkInstanceId,
+    pub url: String,
+    pub version: String,
+    pub genesis_block: BlockPointer,
+    pub adapter: Arc<dyn EthereumAdapterTrait>,
+}
+
+pub fn parse_descriptor(descriptor: &str) -> Result<Descriptor, Error> {
+    if descriptor.starts_with("wss://")
+        || descriptor.starts_with("http://")
+        || descriptor.starts_with("https://")
+    {
+        return Err(format_err!(
+            "Is your Ethereum node string missing a chain name? \
+                     Try 'mainnet:' + the Ethereum node URL."
+        ));
+    }
+
+    // Parse string (format is "NETWORK_NAME:URL")
+    let split_at = descriptor.find(':').ok_or_else(|| {
+        format_err!(
+            "A chain name must be provided alongside the \
+                     Ethereum node location. Try e.g. 'mainnet:URL'."
+        )
+    })?;
+
+    let (name, loc_with_delim) = descriptor.split_at(split_at);
+    let loc = &loc_with_delim[1..];
+
+    if name.is_empty() {
+        Err(format_err!("Ethereum chain name cannot be an empty string"))?;
+    }
+
+    if loc.is_empty() {
+        Err(format_err!("Ethereum node URL cannot be an empty string"))?;
+    }
+
+    Ok(Descriptor {
+        id: NetworkInstanceId {
+            network: "ethereum".into(),
+            name: name.into(),
+        },
+        url: loc.into(),
+    })
+}
+
+pub async fn connect<'a, MR>(options: ConnectOptions<'a, MR>) -> Result<Connection, Error>
+where
+    MR: MetricsRegistry,
+{
+    let ConnectOptions {
+        descriptor,
+        logger,
+        metrics_registry,
+    } = options;
+
+    // Establish an Ethereum adapter for this chain
+    let (transport_event_loop, transport) = Transport::new_rpc(descriptor.url.as_str());
+    let adapter_metrics = Arc::new(ProviderEthRpcMetrics::new(metrics_registry));
+    let adapter = Arc::new(EthereumAdapter::new(transport, adapter_metrics));
+
+    // Connect to the node and obtain the network ID and version
+    let identifier = adapter.net_identifiers(&logger).compat().await?;
+
+    // If we drop the event loop the transport will stop working.
+    // For now it's fine to just leak it.
+    std::mem::forget(transport_event_loop);
```

Comment:
Would it be better to store it in the `Connection` so it gets dropped when the connection gets dropped? That way, there's no leak.